cukes-rest/src/main/java/lv/ctco/cukesrest/internal/ResponseFacade.java

```diff
@@ -38,8 +45,15 @@ public void doRequest(String httpMethod, final String url) throws Exception {
             TimeUnit unit = awaitCondition.getWaitTime().getUnitDict().getTimeUnit();
 
             // TODO Fix
-            with().pollInterval(intervalTime, intervalUnit).
-                await().atMost(waitTime, unit).until(doRequest(url, method), awaitCondition.getResponseMatcher());
+
+            if (awaitCondition.getFailureMatcher() != null) {
```

Comment:
Let's remove if/else. Extract responseMatcher to property and place it into Awaitility method